lib/chef/provider.rb

```diff
@@ -390,9 +390,6 @@ def action_#{name}
           end
         end
       end
-
-      require "chef/dsl/recipe"
-      include Chef::DSL::Recipe
```

Comment:
I guess someone could have discovered that if you only inherit from Chef::Provider but you declare `use_inline_resources` that it mixes in the DSL.  But `use_inline_resources` was only defined on LWRPBase before 12.5 so that usage would be broken on 11.0-12.4.  We've also never documented that and always documented inheriting from LWRPBase in order to get the DSL.  So this would be reverting an unintended undocumented feature to the behavior of prior revisions, which should not be semver violating.

If it causes chaos we can back this out and introduce another directive that means the same thing but avoids mixin in the DSL, but I'd prefer not to go there until we prove that it actually is a problem (we are accumulating a large amount of cruft trying to solve backcompat issues that aren't issues anyone has ever hit before, and its making the code a lot harder to follow).